File: ChinaPost/camerapicklibrary/src/main/java/com/github/zhangyang/camera_picker/exception/ContentException.java
package com.github.zhangyang.camera_picker.exception;

/**
 * 
 * @项目名：AroundYou 
 * @类名称：NetCodeException   
 * @类描述：   服务端异常
 * @创建人：HXF   
 * @修改人：    
 * @创建时间：2015-12-17 上午11:28:10  
 * @version    
 *
 */
@SuppressWarnings("serial")
public class ContentException extends Exception {

	// 错误的代码的叙述
	private int errorCode;
    private String errorContent;//异常描述

	public ContentException() {
		super();
	}

	public ContentException(int errorCode) {
		super();
		this.errorCode = errorCode;
	}

	public ContentException(String errorContent) {
		super(errorContent);
		this.errorContent = errorContent;
	}

	public ContentException(int errorCode, String errorContent) {
		this.errorCode = errorCode;
		this.errorContent = errorContent;
	}

	public ContentException(String detailMessage, Throwable throwable) {
		super(detailMessage, throwable);
	}

	public ContentException(Throwable throwable) {
		super(throwable);
	}

	public int getErrorCode() {
		return errorCode;
	}

	public void setErrorCode(int errorCode) {
		this.errorCode = errorCode;
	}

	public String getErrorContent() {
		return errorContent;
	}

	public void setErrorContent(String errorContent) {
		this.errorContent = errorContent;
	}
}

File: ChinaPost/baselibrary/src/main/java/com/suntray/chinapost/baselibrary/exception/ContentException.kt
package com.suntray.chinapost.baselibrary.exception

/**
 * Created by zhangyang on 2018/5/17 17:15.
 * version 1
 */
class ContentException : Exception {

        // 错误的代码的叙述
        var errorCode: Int = 0
        var errorContent = ""//异常描述
        var code: String? = null
        var obj: Any?=null


        constructor() : super() {}

        constructor(errorCode: Int) : super() {
            this.errorCode = errorCode
        }

        constructor(errorContent: String) : super(errorContent) {
            this.errorContent = errorContent
        }


        /**
         * 构造函数
         * @param Code
         * *
         * @param obj
         */
        constructor(Code: String, `obj`: Any, errorContent: String) : super(Code) {
            this.code = Code
            this.`obj` = `obj`
            this.errorContent = errorContent
        }

        /**
         * 构造函数
         * @param Code
         * *
         * @param obj
         */
        constructor(Code: String, `obj`: Any) : super(Code) {
            this.code = Code
            this.`obj` = `obj`
            this.errorContent = errorContent
        }

        constructor(errorCode: Int, errorContent: String) {
            this.errorCode = errorCode
            this.errorContent = errorContent
        }

        constructor(errorContent: String, Code: String) {
            this.code = Code
            this.errorContent = errorContent
        }


        constructor(detailMessage: String, throwable: Throwable) : super(detailMessage, throwable) {}

        constructor(throwable: Throwable) : super(throwable) {}
}
